src/saveDocs.ts

```diff
@@ -58,6 +63,18 @@ async function addMetaData(pkg: RawPkg): Promise<FinalPkg> {
     },
   };
 
+  const hasFewDownloads =
+    final.downloadsLast30Days <= config.alternativeNamesNpmDownloadsThreshold &&
+    final.jsDelivrHits <= config.alternativeNamesJsDelivrHitsThreshold;
+
+  const addPopularAlternativeNames =
+    final.popular ||
+    (!final.isDeprecated && !final.isSecurityHeld && !hasFewDownloads);
+
+  if (addPopularAlternativeNames) {
+    final._searchInternal.popularAlternativeNames = [];
```

Comment:
We also can do the condition before creating final:

```js
const popularAlternativeNames = [];

const hasFewDownloads =
  final.downloadsLast30Days <= config.alternativeNamesNpmDownloadsThreshold &&
  final.jsDelivrHits <= config.alternativeNamesJsDelivrHitsThreshold;

const addPopularAlternativeNames =
  final.popular ||
  (!final.isDeprecated && !final.isSecurityHeld && !hasFewDownloads);

if (addPopularAlternativeNames) {
  popularAlternativeNames.push(...pkg._searchInternal.alternativeNames);
}

const final = {
  // ...
  _searchInternal: {
    ...pkg._searchInternal,
    popularAlternativeNames,
  },
};
```

or in a function:

```js
function s() {
  const popularAlternativeNames = getPopularAlternativeNames(pkg);
  const final = {
    // ...
    _searchInternal: {
      ...pkg._searchInternal,
      popularAlternativeNames,
    },
  };
}

function getPopularAlternativeNames(pkg) {
  const hasFewDownloads =
    pkg.downloadsLast30Days <= config.alternativeNamesNpmDownloadsThreshold &&
    pkg.jsDelivrHits <= config.alternativeNamesJsDelivrHitsThreshold;

  const addPopularAlternativeNames =
    pkg.popular ||
    (!pkg.isDeprecated && !pkg.isSecurityHeld && !hasFewDownloads);

  if (addPopularAlternativeNames) {
    return pkg._searchInternal.alternativeNames;
  }

  return [];
}
```

Reply:
We could but then each property is on a different object and some null checks are needed, so it looks like this:

```js
  const popularAlternativeNames: string[] = [];

  const hasFewDownloads =
      (download?.downloadsLast30Days || 0) <= config.alternativeNamesNpmDownloadsThreshold &&
      hit.jsDelivrHits <= config.alternativeNamesJsDelivrHitsThreshold;

  const addPopularAlternativeNames =
      download?.popular ||
      (!pkg.isDeprecated && !pkg.isSecurityHeld && !hasFewDownloads);

  if (addPopularAlternativeNames) {
    popularAlternativeNames.push(...pkg._searchInternal.alternativeNames);
  }
```

Reply:
fine as-is then 👍